pymongo/pyopenssl_context.py

```diff
@@ -107,6 +107,9 @@ def _call(self, call, *args, **kwargs):
             try:
                 return call(*args, **kwargs)
             except BLOCKING_IO_ERRORS as exc:
+                # Check for closed socket.
+                if self.fileno() == -1:
+                    raise
```

Comment:
When this case happens what traceback does the user see? I ask because we normally hide BLOCKING_IO_ERRORS type errors from the user and instead raise _socket.timeout like we do below.

Reply:
I agree we should override the error, updated.